src/node_messaging.cc

```diff
@@ -1374,12 +1403,185 @@ static void RegisterExternalReferences(ExternalReferenceRegistry* registry) {
   registry->Register(MessagePort::Drain);
   registry->Register(MessagePort::ReceiveMessage);
   registry->Register(MessagePort::MoveToContext);
+  registry->Register(BroadcastChannel::New);
+  registry->Register(BroadcastChannel::Close);
+  registry->Register(BroadcastChannel::PostMessage);
   registry->Register(SetDeserializerCreateObjectFunction);
 }
 
+static Mutex ports_mutex_;
+
 }  // anonymous namespace
 
+BroadcastChannelService::~BroadcastChannelService() {}
+
+std::unique_ptr<MessagePortData> BroadcastChannelService::Get(
+    std::string channel,
+    MessagePortData* other) {
+  Mutex::ScopedLock lock(ports_mutex_);
+  BroadcastPort* port = nullptr;
+  auto i = ports_.find(channel);
+  if (i == ports_.end()) {
+    auto e = ports_.emplace(channel, new BroadcastPort());
+    if (e.second)
+      port = e.first->second.get();
+  } else {
+    port = i->second.get();
+  }
+  return port->Get(other);
+}
+
+void BroadcastChannelService::BroadcastPort::TriggerAsync(
+    MessagePortData* data) {
+
+  // If the MessagePortData is not entangled with anything, ignore
+  if (!data->is_entangled_with())
+    return;
+
+  size_t processing_limit = data->processing_limit(1000);
+  if (processing_limit == 0)
+    return;
+
+  std::set<MessagePortData*> ports;
+  {
+    Mutex::ScopedLock lock(ports_mutex_);
+    std::copy_if(
+        ports_.begin(),
+        ports_.end(),
+        std::inserter(ports, ports.end()),
+        [data](MessagePortData* port) {
+          // Do not deliver the message to the BroadcastChannel
+          // instance that sent it.
+          return !data->is_entangled_with(port);
+        });
+  }
+  if (ports.size() == 0)
+    return;
+
+  while (processing_limit > 0) {
+    processing_limit--;
+    Message message = data->next_message();
+    for (auto port : ports) {
+      port->AddToIncomingQueue(std::move(message.CloneMainMessageOnly()));
```

Comment:
That's there currently precisely because of this question :) I need to test more of what the browsers do in various cases as the spec wasn't super clear on this point.